csrc/host_ir/lower.cpp

```diff
@@ -296,7 +302,10 @@ std::vector<Communication*> lowerCommunication(Expr* c) {
   return comms;
 }
 
-bool isLowerableToCommunication(Expr* expr) {
+bool HostIrLower::canLower(Expr* expr) {
+  if (!isResharding(expr)) {
+    return true;
+  }
```

Comment:
This apparently the behavior and therefore not a pure refactoring. I'm not requesting a breakdown, but can you teach me why this check didn't exist before and/or why we need it now?

Reply:
I don't think there is any change of behavior in the current places `canLower` is called. However you are right that `isLowerableToCommunication` became `HostIrLower::canLower` and that those methods don't return the same on a non-resharding expr.

The motivation is that Host Ir Lowering will no longer deal with pure communication, but also with compute. Any non-resharding op can be lowered to a `PostOnStream(HostUnit(Fusion))`, or into a Host Op like `MatmulOp` (which will be expr-evaluated). So it is natural that `HostIrLower::canLower` returns `true` on a non-resharding op.

I btw recall that in the coming PR, we will add lowering of distributed matmuls, which embeds both compute and comm.

As a side note, `canLower` should probably eventually be used in the Host segmenter, to know if a segment is valid (the same `canSchedule` is used).